internal/resources/aws/nat_gateway.go

```diff
@@ -6,16 +6,28 @@ import (
 )
 
 type NATGatewayArguments struct {
-	Address string `json:"address,omitempty"`
-	Region  string `json:"region,omitempty"`
+	Address *string `json:"address,omitempty"`
+	Region  *string `json:"region,omitempty" infracost_usage:"region,us-east1,Region where gateway is located,infracost"`
 
-	MonthlyDataProcessedGB *float64 `json:"monthlyDataProcessedGB,omitempty"`
+	MonthlyDataProcessedGB *float64 `json:"monthlyDataProcessedGB,omitempty" infracost_usage:"monthly_data_processed_gb,0,Monthly data processed by the NAT Gateway in GB,infracost,terraform"`
 }
 
-func (args *NATGatewayArguments) PopulateUsage(u *schema.UsageData) {
+func (args *NATGatewayArguments) PopulateArgs(u *schema.UsageData) {
+	address := strPtr("nat_gateway") // TODO: Better default value?
+	region := strPtr("")             // TODO: FIXME: A default value?
+	monthlyDataProcessedGB := floatPtr(0.0)
```

Comment:
Is there anyway to use the info in the args tags to do this in a generic method?

Reply:
Yes sure, I think we can do it.

Reply:
I'll try to change the code to see if it works.